### DISCUSSION TITLE
Programming with the editor in C language

### DISCUSSION THREAD

**Discussion by WOLFxxxxxx** [luis605/Lit-Engine-OLD#41] [Ideas]
Programming in the editor in C instead of PY

The PY language is slow to implement and not good for games

**Comment by luis605** [comment]
I agree. While Python is continuously improving, C++ remains a better choice for certain tasks, though using it for scripting presents challenges.

Engines focus on user experience, and lower-level languages like C++ are more complex, making Unreal Engine harder for scripting. Game engines often allow hot reloads, but waiting for C++ to compile can be frustrating (I had a bad experience with Unity). Performance is critical for engine core functions (e.g., rendering, physics), but game scripts often just call engine functions which is fast. Many engines, like Panda3D, use Python for most scripting and reserve C++ for performance-critical parts, so you can still code in C++ but also use Python. Python's ease of memory management and cross-platform compatibility makes it more appealing, while C++ lacks dynamic features like reflection, making it less user-friendly for scripting.

C++ is expected to be implemented into the scripting module in this version, but solving this issues is important for a better user experience.

**Reply by Kirandeep-Singh-Khehra** [reply]
Hi @luis605,

Instead of using scripting like functionality if we can have something like godot. Show 3d editor in engine and run the game in separate window when run button is pressed and send logs and diagnostics back to IDE. It can help to support a lot of languages.

But if you want to go with scripting like functionality then i guess compiling scripts to a `dll` or `so` and loading it at runtime can have behavior similar to hot-reloading.

And ahead of time compilation of files(to object(`.o`) files using `-c` flag) and then compile all when run button is pressed can also reduce a lot of time.

A mix up like some languages use hot-reloading others don't. or have something like language plugins to support a language.

I'll try this once and check if all of my yapping works or not.

> Tried godot a long time ago so i am not aware if such behavior still exists.

**Comment by WOLFxxxxxx** [comment]
Is it possible to add the mojo language as it is based on py?

https://github.com/andresnowak/raylib-mojo
![Screenshot_٢٠٢٤-١٠-١٢-٢٠-١٦-٢٢-٣٦٩_com android chrome-edit](https://github.com/user-attachments/assets/06ea42b9-b6ce-41e4-aec2-a70abea8dac0)

**Reply by luis605** [reply]
Maybe in the future. The current version being developed has already many desired features to be implemented.
